scripts/windows/agent_diagnostics.ps1

```diff
@@ -32,17 +32,19 @@ function zipjc {
 
 function ziplog {
     # Zip the log files.
-    $LOGFILE="jcagent.log"
-    cp $JCLOG\$LOGFILE $WRITEPATH\$LOGFILE
-    Compress-Archive -Path $WRITEPATH\$LOGFILE -Update -DestinationPath $WRITEPATH\$ZIPFILE
-    Remove-Item -Path $WRITEPATH\$LOGFILE
-    $ziplog_out+=("`tjcagent.log has been added to $ZIPFILE.zip")
+    $fileArray=(Get-ChildItem -Include jc*, Jump* -Name)
+    foreach($LOGFILE in $fileArray) {
+        Copy-Item $JCLOG\$LOGFILE $WRITEPATH\$LOGFILE
```

Comment:
Why do we need to copy the file before adding it to the archive?

Reply:
When attempting to write log files housed in 'c:\windows\temp' to zip with compress-archive, it fails with an error stating:

'The process cannot access the file 'C:\Windows\Temp\jcagent.log' because it is being used by another process.'

The workaround for this is to copy to the needed files to a temp location and compress from there.